lib/src/extensions.dart

```diff
@@ -148,6 +148,44 @@ extension AstNodeExtension on AstNode? {
     }
     return null;
   }
+
+  /// Builds the list resulting from traversing the node in DFS and does not
+  /// include the node itself.
+  ///
+  /// It excludes the nodes for which the [excludeCriteria] returns true. If
+  /// [excludeCriteria] is not provided, all nodes are included.
+  Iterable<AstNode> traverseNodesInDFS({AstNodePredicate? excludeCriteria}) {
```

Comment:
Not necessary, because cleaning up this method wasn't the goal, but ...

Given the potential size of the list being returned, I wonder whether this would be a place where it would be worth the overhead to use `sync*`.

Reply:
Yeah I wondered that too.

I think I'll leave this as is, and look at the use cases in linter, to see how many of them immediately call something like `firstWhere`, etc. I don't think it would be out of bounds to maybe add a new method here for different use cases.